src/views/list.ejs

```diff
@@ -1,80 +1,107 @@
+  <title>StorageVirtualization</title>
+  <link rel="stylesheet" href="/stylesheets/style.css" />
+  <script src="http://code.jquery.com/jquery-1.8.3.min.js"></script>
+  <meta name="viewport" content="width=device-width, initial-scale=1" />
+</head>
 
-    <p id="dlink"></p>
+<body>
+  <%- include partials/nav %>
+  <style>
+    table {
+      border-collapse: collapse;
+    }
 
-    <script>
-      function getDownloadLink(hash) {
-        console.log("hash is", hash);
-        document.getElementById("dlink").innerHTML =
-          "Please wait while the link loads..";
-        $.ajax({
-          type: "GET",
-          url: "/getServerToDownloadFrom?hash=" + hash,
-          success: function(response) {
-            console.log("sresponse", response);
-            document.getElementById("dlink").innerHTML =
-              "<a href=" + response.url + ">Download Link</a>";
-          },
-          error: function(response) {
-            console.log("fresponse", response);
-          }
-        });
-      }
-    </script>
-  </body>
-</html>
+    table,
+    th,
+    td {
+      border: 1px solid black;
+    }
+  </style>
+  <% if(message) {%>
+  <p style="color: red"><%= message %></p>
+  <% } %>
+  <table id="table1">
+    <tr>
+      <td>
+        Filename
+      </td>
+      <td>
+        Size(in bytes)
+      </td>
+      <td>Date Uploaded</td>
+      <td>
+        Hash
+      </td>
+      <td>
+        Action
+      </td>
+    </tr>
+    <% map.forEach(function(file, i){ %>
+    <tr>
+      <td><%= file.name %></td>
+      <td><%= file.size %></td>
+      <td><%= file.date %></td>
+      <td><%= file.hash %></td>
+      <td>
+        <button onclick="getDownloadLink('<%= file.hash %>')">
+          Generate Download Link
+        </button>
+        <button onclick="Delete('<%= file.hash %>')">
+          Delete File
+        </button>
+
+      </td>
+    </tr>
+    <% }); %>
+  </table>
+
+  <p id="dlink"></p>
+  <p id="delete"></p>
+
+  <script>
+    function getDownloadLink(hash) {
+      console.log("hash is", hash);
+      document.getElementById("dlink").innerHTML =
+        "Please wait while the link loads..";
+      $.ajax({
+        type: "GET",
+        url: "/getServerToDownloadFrom?hash=" + hash,
+        success: function (response) {
+          console.log("sresponse", response);
+          document.getElementById("dlink").innerHTML =
+            "<a href=" + response.url + ">Download Link</a>";
+        },
+        error: function (response) {
+          console.log("fresponse", response);
+        }
+      });
+    }
+    function Delete(hash) {
+      console.log("hash is", hash);
+      document.getElementById("delete").innerHTML =
+        "Deleting..";
+      $.ajax({
+        type: "GET",
+        url: "/delete?hash=" + hash,
+        success: function (response) {
+          console.log("sresponse", response);
+          document.getElementById("delete").innerHTML =
+            "<p>" + response.url + "</p>";
```

Comment:
response.url will be empty, so you can update this to rather display the message